src/main/groovy/com/palantir/gradle/gitversion/JGitDescribe.groovy

```diff
@@ -0,0 +1,101 @@
+package com.palantir.gradle.gitversion
+
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.internal.storage.file.FileRepository
+import org.eclipse.jgit.lib.Constants
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.PersonIdent
+import org.eclipse.jgit.lib.Ref
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevTag
+import org.eclipse.jgit.revwalk.RevWalk
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class JGitDescribe implements GitDescribe {
```

Comment:
I think we need to give future readers some javadoc to help them understand why we built this - specifically that shelling out to `git` is a bit unsafe because older versions of git don't support `--first-parent` 

https://github.com/palantir/gradle-git-version/issues/46

Reply:
Yes. I'll also add comment in GitVersionPlugin.gitDescribe to explain running both native and jgit